packages/shared/src/lib/core/isc/utils/buildIscAssets.ts

```diff
@@ -1,14 +1,14 @@
+import { IscAssets } from '@core/isc/types'
 import { IIscChain } from '@core/network/interfaces'
-import { AssetType } from '../enums'
-import { ILayer2AssetAllowance } from '../interfaces'
-import { TransferredAsset } from '../types'
+import { AssetType } from '../../layer-2/enums'
+import { TransferredAsset } from '../../layer-2/types'
 
-export function buildAssetAllowance(
+export function buildIscAssets(
     iscChain: IIscChain,
     transferredAsset: TransferredAsset,
     storageDepositRequired?: bigint
-): ILayer2AssetAllowance {
-    const baseTokens = String(storageDepositRequired ?? 0)
+): IscAssets {
+    const baseTokens = BigInt(storageDepositRequired ?? 0).toString()
```

Comment:
what difference does this make?

Reply:
Not sure, I think I tried to use bigints everywhere but it didn't work, thats because we have two types of places that this IscAssets is used, in web3.js paramaters which will accept bigints, but also in the parameters in the call function of the Magic contract which does not do the same conversions as web3.js

Reply:
So I first changed to bigint, then realised it should be a string